lib/player_ai_minimax.rb

```diff
@@ -0,0 +1,57 @@
+class PlayerAiMinimax
+  attr_reader :mark
+
+  def initialize(mark, rules)
+    @mark = mark
+    @rules = rules
+  end
+
+  def play_move(game_state)
+    board = game_state.game_board
+    return 0 if board.empty?
+    return 4 if opponent_played_corner_for_first_move(board)
+    @current_player = game_state.player
+    @non_current_player = game_state.opponent
+    minimax(board, game_state.player, game_state.opponent)
+    @choice
+  end
+
+  private
+  def opponent_played_corner_for_first_move(board)
+    corners = [0,2,6,8]
+    if (board.available_moves.size == 8)
```

Comment:
look at the Enumerable module for the `any?` method.

Could probably do something like: `corners.any?{ |corner| board.spaces[corner] != ' ' }`